ios/RNCCameraRollManager.m

```diff
@@ -286,34 +296,47 @@ static void RCTResolvePromise(RCTPromiseResolveBlock resolve,
-          foundAfter = YES;
-        }
-        return; // skip until we get to the first one
+      NSString *_Nullable originalFilename = NULL;
+      PHAssetResource *_Nullable resource = NULL;
+      NSNumber* fileSize = [NSNumber numberWithInt:0];
+      
+      if (includeFilename || includeFileSize || [mimeTypes count] > 0) {
+        // Get underlying resources of an asset - this includes files as well as details about edited PHAssets
+        // This is required for the filename and mimeType filtering
+        NSArray<PHAssetResource *> *const assetResources = [PHAssetResource assetResourcesForAsset:asset];
+        resource = [assetResources firstObject];
+        originalFilename = resource.originalFilename;
+        fileSize = [resource valueForKey:@"fileSize"];
       }
+      
+      // WARNING: If you add any code to `collectAsset` that may skip adding an
+      // asset to the `assets` output array, you should do it inside this
+      // block and ensure the logic for `collectAssetMayOmitAsset` above is
+      // updated
+      if (collectAssetMayOmitAsset) {
+        if (afterCursor && !foundAfter) {
+          if ([afterCursor isEqualToString:uri]) {
+            foundAfter = YES;
+          }
+          return; // skip until we get to the first one
+        }
 
-      // Get underlying resources of an asset - this includes files as well as details about edited PHAssets
-      NSArray<PHAssetResource *> *const assetResources = [PHAssetResource assetResourcesForAsset:asset];
-      if (![assetResources firstObject]) {
-        return;
-      }
-      PHAssetResource *const _Nonnull resource = [assetResources firstObject];
 
-      if ([mimeTypes count] > 0) {
-        CFStringRef const uti = (__bridge CFStringRef _Nonnull)(resource.uniformTypeIdentifier);
-        NSString *const mimeType = (NSString *)CFBridgingRelease(UTTypeCopyPreferredTagWithClass(uti, kUTTagClassMIMEType));
+        if ([mimeTypes count] > 0 && resource) {
+          CFStringRef const uti = (__bridge CFStringRef _Nonnull)(resource.uniformTypeIdentifier);
+          NSString *const mimeType = (NSString *)CFBridgingRelease(UTTypeCopyPreferredTagWithClass(uti, kUTTagClassMIMEType));
 
-        BOOL __block mimeTypeFound = NO;
-        [mimeTypes enumerateObjectsUsingBlock:^(NSString * _Nonnull mimeTypeFilter, NSUInteger idx, BOOL * _Nonnull stop) {
-          if ([mimeType isEqualToString:mimeTypeFilter]) {
-            mimeTypeFound = YES;
-            *stop = YES;
-          }
-        }];
+          BOOL __block mimeTypeFound = NO;
+          [mimeTypes enumerateObjectsUsingBlock:^(NSString * _Nonnull mimeTypeFilter, NSUInteger idx, BOOL * _Nonnull stop) {
+            if ([mimeType isEqualToString:mimeTypeFilter]) {
+              mimeTypeFound = YES;
+              *stop = YES;
+            }
+          }];
 
-        if (!mimeTypeFound) {
-          return;
+          if (!mimeTypeFound) {
+            return;
+          }
         }
       }
 
```

Comment:
Can't comment on the exact line. So I put it here.
You should move this code block to the end of function. Otherwise, it can't be executed because the last element is not added to list.
```
      // If we've accumulated enough results to resolve a single promise
      if (first == assets.count) {
        *stopAssets = YES;
        stopCollections_ = YES;
        hasNextPage = YES;
        RCTAssert(resolvedPromise == NO, @"Resolved the promise before we finished processing the results.");
        RCTResolvePromise(resolve, assets, hasNextPage);
        resolvedPromise = YES;
        return;
      }
```

Reply:
This code was there before this change, and is working correctly (both before and after the change).

If we've already gotten the number the user has requested, we should not add any more items.

Keeping the code here instead at the end of the callback block also ensures that it runs correctly and returns 0 items if `first` is set to be 0.